x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/task/rule_migrations_task_runner.ts

```diff
@@ -192,15 +200,23 @@ export class RuleMigrationTaskRunner {
       } while (true);
 
       migrationTaskTelemetry.success();
+      await this.saveRuleMigrationCompleted({});
       this.logger.info('Migration completed successfully');
     } catch (error) {
       await this.data.rules.releaseProcessing(migrationId);
 
       migrationTaskTelemetry.failure(error);
       if (error instanceof AbortError) {
         this.logger.info('Abort signal received, stopping migration');
+        await this.saveRuleMigrationCompleted({
```

Comment:
Unfortunately, both the manual and automatic stops will execute this code. Let's sync about this.

Reply:
Checkout new changes in [task_client](https://github.com/elastic/kibana/pull/221635/files/cfab544b19ba8262ed93a5663c96e350bde94072#diff-3d83c1ba1229d3f37496f13445b5bb9d6179a373f9d0f17f7edb19cd1cbea61a)